src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,158 @@
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = findByIndex(index);
+        T changedValue = currentNode.item;
+        currentNode.item = value;
+        return changedValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = findByIndex(index);
+        if (currentNode.prev != null && currentNode.next != null) {
+            currentNode.prev.next = currentNode.next;
+            currentNode.next.prev = currentNode.prev;
+        } else if (currentNode == head && currentNode.next != null) {
+            currentNode.next.prev = null;
+            head = currentNode.next;
+        } else if (currentNode == tail && currentNode.prev != null) {
+            currentNode.prev.next = null;
+            tail = currentNode.prev;
+        }
+        size--;
+        return currentNode.item;
     }
 
     @Override
     public boolean remove(T object) {
+        for (int i = 0; i < size; i++) {
+            Node<T> nodeToRemove = findByIndex(i);
+            if ((object == null && nodeToRemove.item == null)
+                    || (object != null && object.equals(nodeToRemove.item))) {
+                if (nodeToRemove.prev != null && nodeToRemove.next != null) {
+                    nodeToRemove.prev.next = nodeToRemove.next;
+                    nodeToRemove.next.prev = nodeToRemove.prev;
+                } else if (nodeToRemove == head && nodeToRemove.next != null) {
+                    nodeToRemove.next.prev = null;
+                    head = nodeToRemove.next;
+                } else if (nodeToRemove == tail && nodeToRemove.prev != null) {
+                    nodeToRemove.prev.next = null;
+                    tail = nodeToRemove.prev;
+                }
+                size--;
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size <= 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of bounds");
+        }
+    }
+
+    private void checkIndexForAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of bounds");
+        }
+    }
+
+    private Node<T> findByIndex(int index) {
+        Node<T> currentByIndex;
+        if (index < size / 2) {
+            currentByIndex = head;
+            for (int i = 0; i < index; i++) {
+                currentByIndex = currentByIndex.next;
+            }
+        } else {
+            currentByIndex = tail;
+            for (int i = size - 1; i > index; i--) {
+                currentByIndex = currentByIndex.prev;
+            }
+        }
+        return currentByIndex;
+    }
```

Comment:
Don't iterate your nodes from the head to the tail or vice versa. Pay attention to the index you receive as an input, you can iterate only the first or second half of the list, depending on the index value. That will boost your lists' performance.